extensions/blocks/recurring-payments/edit.jsx

```diff
@@ -422,117 +513,29 @@ class MembershipsButtonEdit extends Component {
-							instructions={ __(
-								"You'll need to upgrade your plan to use the Payments block.",
-								'jetpack'
-							) }
-						>
-							<Button isSecondary isLarge href={ this.state.upgradeURL } target="_blank">
-								{ __( 'Upgrade your plan', 'jetpack' ) }
-							</Button>
-							{ this.renderDisclaimer() }
-						</Placeholder>
-					</div>
-				) }
-				{ ( connected === API_STATE_LOADING ||
-					this.state.addingMembershipAmount === PRODUCT_FORM_SUBMITTED ) &&
-					! this.props.attributes.planId && (
-						<Placeholder icon={ <BlockIcon icon={ icon } /> } notices={ notices }>
-							<Spinner />
-						</Placeholder>
-					) }
-				{ ! this.state.shouldUpgrade &&
-					! this.props.attributes.planId &&
-					connected === API_STATE_CONNECTED &&
-					products.length === 0 && (
-						<div className="wp-block-jetpack-recurring-payments">
-							<Placeholder
-								icon={ <BlockIcon icon={ icon } /> }
-								label={ __( 'Payments', 'jetpack' ) }
-								notices={ notices }
-							>
-								<div className="components-placeholder__instructions">
-									<p>
-										{ __( 'To use this block, first add at least one payment plan.', 'jetpack' ) }
-									</p>
-									{ this.renderAddMembershipAmount( true ) }
-									{ this.renderDisclaimer() }
-								</div>
-							</Placeholder>
-						</div>
-					) }
-				{ ! this.state.shouldUpgrade &&
-					! this.props.attributes.planId &&
-					this.state.addingMembershipAmount !== PRODUCT_FORM_SUBMITTED &&
-					connected === API_STATE_CONNECTED &&
-					products.length > 0 && (
-						<div className="wp-block-jetpack-recurring-payments">
-							<Placeholder
-								icon={ <BlockIcon icon={ icon } /> }
-								label={ __( 'Payments', 'jetpack' ) }
-								notices={ notices }
-							>
-								<div className="components-placeholder__instructions">
-									<p>
-										{ __(
-											'To use this block, select a previously created payment plan.',
-											'jetpack'
-										) }
-									</p>
-									{ this.renderMembershipAmounts() }
-									<p>{ __( 'Or a new one.', 'jetpack' ) }</p>
-									{ this.renderAddMembershipAmount( false ) }
-									{ this.renderDisclaimer() }
-								</div>
-							</Placeholder>
-						</div>
-					) }
+				{ ! this.isPremiumContentChild && this.renderUpgradeNudges() }
 				{ showControls && inspectorControls }
-				{ ( ( ( this.hasUpgradeNudge || ! this.state.shouldUpgrade ) &&
-					connected !== API_STATE_LOADING ) ||
-					this.props.attributes.planId ) && (
-					<InnerBlocks
-						template={ [
-							[
-								'jetpack/button',
-								{
-									element: 'a',
-									uniqueId: 'recurring-payments-id',
-								},
-							],
-						] }
-						templateLock="all"
-					/>
-				) }
-				{ this.hasUpgradeNudge && connected === API_STATE_NOTCONNECTED && (
-					<div className="wp-block-jetpack-recurring-payments disclaimer-only">
-						{ this.renderDisclaimer() }
-					</div>
-				) }
+
+				<InnerBlocks
```

Comment:
Should we hide the button when you don't have a plan yet, since you can't use the button yet?

Reply:
We are showing the buttons on the front end, but disabling the actions. We don't want to disable front end rendering where possible so that we can build patterns that accurately depict what you will get with a plan. Instead we're showing front end notices above the content. See: https://github.com/Automattic/view-design/issues/95

Reply:
We are showing the buttons on the front end, but disabling the actions. We don't want to disable front end rendering where possible so that we can build patterns that accurately depict what you will get with a plan. Instead we're showing front end notices above the content. See: Automattic/view-design#95